### PR TITLE
PR: Added the Solarized color scheme

### PR DESCRIPTION
**Pull request by grantcarthew**

### CODE REVIEW

**Comment by goanpeca** [comment]
@grantcarthew thanks for this :-p

Could you paste a couple of screenshots for reference?

**Comment by ccordoba12** [comment]
Great addition! Thanks a lot for it :-)

It seems that when you did your changes, you changed the file type of  `spyder/config/main.py` to be executable (i.e. mode 755 on Unix). Please revert that :-)

I have other comments, but I'll do them in my review :-)

File: spyder/config/main.py [hunk 1]
```diff
@@ -420,7 +420,7 @@
             ('color_schemes',
              {
               'names': ['emacs', 'idle', 'monokai', 'pydev', 'scintilla',
-                        'spyder', 'spyder/dark', 'zenburn'],
+                        'spyder', 'spyder/dark', 'zenburn', 'SolarizedDark', 'SolarizedLight'],
```
Comment (ccordoba12) [inline spyder/config/main.py]: I think these should be `solarized/dark` and `solarized`

File: spyder/config/main.py [hunk 2]
```diff
@@ -573,7 +573,45 @@
               'zenburn/comment':    ('#7f9f7f', False, True),
               'zenburn/string':     ('#cc9393', False, False),
               'zenburn/number':     ('#8cd0d3', False, False),
-              'zenburn/instance':   ('#dcdccc', False, True)
+              'zenburn/instance':   ('#dcdccc', False, True),
+              # ---- SolarizedDark ----
+              'solarizeddark/name':        "SolarizedDark",
```
Comment (ccordoba12) [inline spyder/config/main.py]: The name is what appears in Spyder Preferences, so it needs to be `Solarzied Dark`

File: spyder/config/main.py [hunk 3]
```diff
@@ -573,7 +573,45 @@
               'zenburn/comment':    ('#7f9f7f', False, True),
               'zenburn/string':     ('#cc9393', False, False),
               'zenburn/number':     ('#8cd0d3', False, False),
-              'zenburn/instance':   ('#dcdccc', False, True)
+              'zenburn/instance':   ('#dcdccc', False, True),
+              # ---- SolarizedDark ----
+              'solarizeddark/name':        "SolarizedDark",
+              #        Name            Color     Bold  Italic
+              'solarizeddark/background':  '#002b36',
```
Comment (ccordoba12) [inline spyder/config/main.py]: And these should be `solarized/dark/background: ...`, etc

File: spyder/config/main.py [hunk 4]
```diff
@@ -573,7 +573,45 @@
...
+              'solarizeddark/normal':     ('#839496', False, False),
+              'solarizeddark/keyword':    ('#859900', False, False),
+              'solarizeddark/builtin':    ('#6c71c4', False, False),
+              'solarizeddark/definition': ('#268bd2', True, False),
+              'solarizeddark/comment':    ('#586e75', False, True),
+              'solarizeddark/string':     ('#2aa198', False, False),
+              'solarizeddark/number':     ('#cb4b16', False, False),
+              'solarizeddark/instance':   ('#b58900', False, True),
+              # ---- SolarizedLight ----
+              'solarizedlight/name':        "SolarizedLight",
```
Comment (ccordoba12) [inline spyder/config/main.py]: This option needs to be `solarized/name: Solarized`

**Comment by grantcarthew** [comment]
Hi guys, to answer @goanpeca, I got the theme from here (screen shots):

https://github.com/pshchelo/spyder-color-solarized

Thanks to @pshchelo for your work creating the scheme. I am just starting to use spyder for machine learning and was missing the solarized scheme. Put it in from the above instructions and thought, this should be a default option.

>It seems that when you did your changes, you changed the file type of spyder/config/main.py to be executable (i.e. mode 755 on Unix). Please revert that :-)

I didn't realize the permissions changed. What is the default for this file?

I'll look at the inline comments now.

File: spyder/config/main.py (hunk 1 above)
Comment (grantcarthew) [inline spyder/config/main.py]: Yes, you're right.

**Comment by goanpeca** [comment]
Hi again, the default should be `644`

**Comment by grantcarthew** [comment]
Hows that?

**Comment by goanpeca** [comment]
`chmod -R 644 spyder/config/main.py`

or 

`chmod -R 644 main.py` if inside the config folder

?

**Comment by grantcarthew** [comment]
It's showing `-rw-r--r--` here. After I changed it.

**Comment by goanpeca** [comment]
It seems to be ok now @ccordoba12 ?

**Comment by goanpeca** [comment]
@ccordoba12 so this is ready?

**Comment by ccordoba12** [comment]
We can backport this (it's a simple) so we please leave it for 3.1 :-)

File: spyder/config/main.py [hunk 5]
```diff
@@ -420,7 +420,7 @@
             ('color_schemes',
              {
               'names': ['emacs', 'idle', 'monokai', 'pydev', 'scintilla',
-                        'spyder', 'spyder/dark', 'zenburn'],
+                        'spyder', 'spyder/dark', 'zenburn', 'solarized', 'solarized/dark'],
```
Comment (ccordoba12) [inline spyder/config/main.py]: Sorry for asking you to change this again, but I looked at several editors and IDEs, and use "Solarized Light" as the name for the light version of Solarized. So please all variables to `solarized/light`

Comment (ccordoba12) [inline spyder/config/main.py]: Thanks!

**Comment by grantcarthew** [comment]
How's that?

**Comment by ccordoba12** [comment]
Nice, great contribution! Merging now :-)

**Pull request merged**